### PR TITLE
bazel: Update for use with Fujitsu compiler

### PR DESCRIPTION
**Pull request by ghost**
Add 2 patches to use with Fujitsu compiler.

### CODE REVIEW

File: var/spack/repos/builtin/packages/bazel/package.py [hunk 1]
```diff
@@ -128,6 +128,11 @@ class Bazel(Package):
     patch('compile-0.4.patch',  when='@0.4:0.5')
     patch('compile-0.3.patch',  when='@:0.3')
 
+    #for fcc
+    patch('patch_for_fcc.patch', when='@0.29.1:%fj')
+    patch('patch_for_fcc2.patch', when='@0.25:%fj')
+    parallel = False
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/bazel/package.py]: Did you encounter some problems while building in parallel? I've never had a problem, but I know this package takes a long time to compile, so I don't want to build in serial unless we have to.

File: var/spack/repos/builtin/packages/bazel/patch_for_fcc2.patch [hunk 2]
```diff
@@ -0,0 +1,11 @@
+--- a/src/main/java/com/google/devtools/build/lib/bazel/rules/BazelRuleClassProvider.java
++++ b/src/main/java/com/google/devtools/build/lib/bazel/rules/BazelRuleClassProvider.java
+@@ -185,7 +185,7 @@ public class BazelRuleClassProvider {
+ 
+         Map<String, String> spackEnv = System.getenv();
+         for (String envName : spackEnv.keySet()) {
+-          if (envName.startsWith("SPACK_")) {
++          if ((envName.startsWith("SPACK_")) || (envName.equals("fcc_ENV")) || (envName.equals("FCC_ENV"))) {
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/bazel/patch_for_fcc2.patch:8]: I think you'll need to do something similar for older versions as well.

File: var/spack/repos/builtin/packages/bazel/package.py (hunk 1 above)
Comment (ghost) [reply var/spack/repos/builtin/packages/bazel/package.py]: We encountered the problem as follows while building with Fujitsu compiler. 
The missing file differed each time. 

```
  >> 1363    ERROR: /tmp/ihara/spack-stage/spack-stage-bazel-0.29.1-lmznhxdrwncyrhhmvalccpvzi25ggvpr/spack-src/thi
             rd_party/ijar/BUILD:74:1: error while parsing .d file: /tmp/bazel_tni6ciH6/out/execroot/io_bazel/baze
             l-out/aarch64-opt/bin/third_party/ijar/_objs/zipper/zip_main.d (No such file or directory)
```

This problem solved to build in serial.
We want to build in serial only when we build with Fujitsu compiler. 
Therefore, I will modify as follows .What do you think of this idea?

```
    def bootstrap(self, spec, prefix):
+       if self.spec.satisfies('%fj'):
+           parallel = False
        bash = which('bash')
        bash('./compile.sh')
```

Comment (adamjstewart) [reply var/spack/repos/builtin/packages/bazel/package.py]: You could also do something like:
```python
@property
def parallel(self):
    return not self.spec.satisfies('%fj')
```
I'm not sure if setting `parallel` inside a function will propagate outside of the function.

Comment (ghost) [reply var/spack/repos/builtin/packages/bazel/package.py]: Thank you for your advice.
I fixed that according to your advice.

File: var/spack/repos/builtin/packages/bazel/patch_for_fcc2.patch (hunk 2 above)
Comment (ghost) [reply var/spack/repos/builtin/packages/bazel/patch_for_fcc2.patch:8]: I tried to fix something similar for 0.24.1 or less. 
However, the following error occurred when I tried build bazel 0.24.1 on Fujitsu Compiler.

```
     386      /opt/FJSVxtclanga/tcsds-1.2.25/bin/fcc -U_FORTIFY_SOURCE -fstack-
            protector -Wall -Wthread-safety -Wself-assign -fcolor-diagnostics -
            fno-omit-frame-pointer -g0 -O2 '-D_FORTIFY_SOURCE=1' -DNDEBUG -ffun
            ction-sections -fdata-sections -MD -MF bazel-out/aarch64-opt/bin/th
            ird_party/zlib/_objs/zlib/adler32.d '-frandom-seed=bazel-out/aarch6
            4-opt/bin/third_party/zlib/_objs/zlib/adler32.o' -iquote . -iquote
            bazel-out/aarch64-opt/genfiles -iquote bazel-out/aarch64-opt/bin -i
            system third_party/zlib -isystem bazel-out/aarch64-opt/genfiles/thi
            rd_party/zlib -isystem bazel-out/aarch64-opt/bin/third_party/zlib -
            w '-Dverbose=-1' -no-canonical-prefixes -Wno-builtin-macro-redefine
            d '-D__DATE__="redacted"' '-D__TIMESTAMP__="redacted"' '-D__TIME__=
            "redacted"' -c third_party/zlib/adler32.c -o bazel-out/aarch64-opt/
            bin/third_party/zlib/_objs/zlib/adler32.o)
     387    Execution platform: @bazel_tools//platforms:host_platform
     388    [2 / 7] checking cached actions
     389
  >> 390    clang: error: unknown argument: '-fstack-protector'
     391    [2 / 7] checking cached actions
```

Therefore, I will set conflicts method as follows .What do you think of this idea ?

```
    conflicts('%fj', when='@:0.24.1',
          msg='Fujitsu Compiler can not build 0.24.1 or less, '
              'please use a newer release.')
```

**Review by adamjstewart — APPROVED** [review]

**Pull request merged**